src/main/scala/org/zalando/spearheads/innkeeper/oauth/OAuthDirectives.scala

```diff
@@ -23,24 +24,57 @@ trait OAuthDirectives {
   def authenticationToken(requestDescription: String): Directive1[String] =
     headerValue(authorization()) | reject(CredentialsMissingRejection(requestDescription))
 
-  def authenticate(token: String, requestDescription: String, authService: AuthService): Directive1[AuthenticatedUser] =
-    authService.authenticate(token) match {
-      case Success(authUser) => provide(authUser)
-      case Failure(ex) => {
-        logger.error(s"Authentication failed with exception: ${ex.getMessage}")
-        reject(CredentialsRejectedRejection(requestDescription))
+  def authenticate(token: String, requestDescription: String, authUserToRoute: AuthenticatedUser => Route)(implicit authService: AuthService, executionContext: ExecutionContext): Route = {
+
+    onComplete(authService.authenticate(token)) {
+      case Success(userResult) => userResult match {
+        case ServiceResult.Success(team) => {
+          authUserToRoute(team)
+        }
+        case ServiceResult.Failure(Ex(ex)) => {
+          logger.error(s"OAuthService failed with exception $ex")
```

Comment:
Just making sure that you've omitted including ex in the error call (i.e. `error(ex, "")` which includes the exception, not just message then)